File: frontend/src/lib/format.ts
export function formatBytes(b: number): string {
  if (b === 0) return '0 B';
  const units = ['B', 'KB', 'MB', 'GB', 'TB', 'PB'];
  const k = 1024;
  const i = Math.floor(Math.log(b) / Math.log(k));
  return parseFloat((b / Math.pow(k, i)).toFixed(1)) + ' ' + units[i];
}

export function formatBytesPerSec(b: number): string {
  return formatBytes(b) + '/s';
}

export function formatUptime(sec: number, units?: { d?: string; h?: string; m?: string; s?: string }): string {
  const u = { d: 'd', h: 'h', m: 'm', s: 's', ...units };
  if (sec < 60) return sec + u.s;
  if (sec < 3600) return Math.floor(sec / 60) + u.m;
  const days = Math.floor(sec / 86400);
  const hours = Math.floor((sec % 86400) / 3600);
  if (days > 0) return days + u.d + ' ' + hours + u.h;
  return Math.floor(sec / 3600) + u.h + ' ' + Math.floor((sec % 3600) / 60) + u.m;
}

File: model/format.go
package model

import "fmt"

// FormatBytes converts bytes to human-readable string (e.g., "1.5 GB")
func FormatBytes(b uint64) string {
	const unit = 1024
	if b < unit {
		return fmt.Sprintf("%d B", b)
	}
	div, exp := uint64(unit), 0
	for n := b / unit; n >= unit; n /= unit {
		div *= unit
		exp++
	}
	return fmt.Sprintf("%.1f %cB", float64(b)/float64(div), "KMGTPE"[exp])
}

// FormatBytesPerSec converts bytes/sec to human-readable string (e.g., "3.2 MB/s")
func FormatBytesPerSec(b uint64) string {
	return FormatBytes(b) + "/s"
}

// FormatUptime converts seconds to human-readable string (e.g., "3d 12h")
func FormatUptime(sec uint64) string {
	if sec < 60 {
		return fmt.Sprintf("%ds", sec)
	}
	if sec < 3600 {
		return fmt.Sprintf("%dm", sec/60)
	}
	days := sec / 86400
	hours := (sec % 86400) / 3600
	if days > 0 {
		return fmt.Sprintf("%dd %dh", days, hours)
	}
	return fmt.Sprintf("%dh %dm", sec/3600, (sec%3600)/60)
}
